File: 2523/2523.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.lang.*;

public class MyClass {
    
    public static void main(String args[]) throws IOException {
        
          BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
          String linha;
          while ((linha = br.readLine()) != null)
          {
            String conjunto = linha;
            linha = br.readLine();
            int qtdeL = Integer.parseInt(linha);
            linha = br.readLine();
            String pos = linha;
    
            String msg = "";
            String[] list = pos.split(" ");
            for(int j=0; j < qtdeL; j++)
            {
                msg += conjunto.charAt(Integer.parseInt(list[j])-1);
            }
            System.out.println(msg);
          }        
    }
}

File: 2523/2523.js
var lines = ["ABCDEFGHIJKLMNOPQRSTUVWXYZ","4","8 5 12 16","QWERTYUIOPASDFGHJKLZXCVBNM","10","16 3 19 19 9 2 9 4 19 13"];

for(var i=0; i<lines.length; i++){

    var conjunto = lines[i];
    i++;
    var qtdeL = parseInt(lines[i]);
    i++
    var pos = lines[i];

    var msg = "";
    var list = pos.split(' ');
    for(var j=0; j < qtdeL; j++)
    {
        msg += conjunto[parseInt(list[j])-1];
    }
    console.log(msg);
}